pkg/operator/ceph/csi/controller.go

```diff
@@ -140,16 +150,36 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 
 		return reconcile.Result{}, nil
 	} else {
-		for _, cluster := range cephClusters.Items {
+		for i, cluster := range cephClusters.Items {
 			if !cluster.DeletionTimestamp.IsZero() {
-				logger.Debug("ceph cluster is being deleting, no need to reconcile the csi driver")
+				logger.Debugf("ceph cluster %q is being deleting, no need to reconcile the csi driver", cluster.Name)
 				return reconcile.Result{}, nil
 			}
 
 			if !cluster.Spec.External.Enable && cluster.Spec.CleanupPolicy.HasDataDirCleanPolicy() {
-				logger.Debug("ceph cluster has cleanup policy, the cluster will soon go away, no need to reconcile the csi driver")
+				logger.Debugf("ceph cluster %q has cleanup policy, the cluster will soon go away, no need to reconcile the csi driver", cluster.Name)
 				return reconcile.Result{}, nil
 			}
+
+			// Do we have a multus cluster?
+			// If so deploy the plugin holder with the fsid attached
+			if cluster.Spec.Network.IsMultus() {
+				// Load cluster info for later use in updating the ceph-csi configmap
+				clusterInfo, _, _, err := opcontroller.LoadClusterInfo(r.context, r.opManagerContext, cluster.Namespace)
+				if err != nil {
+					// This avoids a requeue with exponential backoff and allows the controller to reconcile
+					// more quickly when the cluster is ready.
+					if errors.Is(err, opcontroller.ClusterInfoNoClusterNoSecret) {
+						return opcontroller.WaitForRequeueIfCephClusterNotReady, nil
+					}
+					return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to populate cluster info for cluster %q", cluster.Name)
+				}
+
+				logger.Debugf("cluster %q is running on multus, deploying the ceph-csi plugin holder", cluster.Name)
+				r.multusClusters = append(r.multusClusters, ClustersInfo{cluster: &cephClusters.Items[i], clusterInfo: clusterInfo})
```

Comment:
There is a new slice with every reconcile, right?

Reply:
Yes?